### PR TITLE
Productspage 03

### LINKED ISSUE #5: feat: Products Page

### PR DESCRIPTION
**Pull request by sadathshariff**
This PR Contains 
1. Added Context for filter
2. Added Filters to Products
3. Fetching Data from DB
4. Created Card, SideNav Components
5. Created UseAxios Hook 
6. Added Products data in DB

### CODE REVIEW

File: src/contexts/filterContext/context.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { useContext, useReducer, useEffect, createContext } from "react";
+import { reducerFunction } from "./reducer";
```
Comment (ttahm3d) [inline src/contexts/filterContext/context.js]: it would be better to name this as filterReducer

File: src/contexts/filterContext/reducer.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      return { ...state, showFastDelivery: action.payload };
+    case "SORT_BY_RATING":
+      return { ...state, rating: action.payload };
+    case "SORT_BY_PRICE_RANGE":
+      return { ...state, priceRange: action.payload };
+    case "FILTER_BY_CATEGORY":
+      const newCategory = state.category.includes(action.payload)
+        ? state.category.filter((p) => p !== action.payload)
+        : [...state.category, action.payload];
+      console.log(newCategory);
```
Comment (ttahm3d) [inline src/contexts/filterContext/reducer.js]: Remove console.log

**Review by ttahm3d — COMMENTED** [review]
Everything looks good. Added a suggestion.

I had a different approach to handling categories than yours nice to know.

File: src/contexts/filterContext/reducer.js (hunk 2 above)
Comment (sadathshariff) [reply src/contexts/filterContext/reducer.js]: removed it

File: src/contexts/filterContext/context.js (hunk 1 above)
Comment (sadathshariff) [reply src/contexts/filterContext/context.js]: Thanks for pointing it out will rename it

File: src/components/Card/ProductCard/ProductCard.css [hunk 3]
```diff
@@ -0,0 +1,9 @@
+.card {
+  width: 19rem;
+  height: max-content;
+  margin: 1rem;
+  background-color: var(--off-white);
+}
+.card-img:hover {
+  cursor: pointer;
+}
```
Comment (kjais1720) [inline src/components/Card/ProductCard/ProductCard.css]: if this is specific to product cards only, then its better to use CSS modules for it

File: src/pages/Products/Products.jsx [hunk 4]
```diff
@@ -1,3 +1,27 @@
+import { Loading, ProductCard } from "../../components/index";
```
Comment (kjais1720) [inline src/pages/Products/Products.jsx:1]: you can directly import the components from `../../components`, anything that's exported from an index.js file of a folder, become default exports for that folder

Comment (sadathshariff) [reply src/pages/Products/Products.jsx:1]: Okay Thanks for pointing it out

File: src/components/Card/ProductCard/ProductCard.css (hunk 3 above)
Comment (sadathshariff) [reply src/components/Card/ProductCard/ProductCard.css]: Yeah added CSS modules for that

**Pull request merged**